### DISCUSSION TITLE
quasar prop spread doesn't work,is't a bug?

### DISCUSSION THREAD

**Discussion by v479038280** [zauberzeug/nicegui#3297] [General]
here is the code
```py
from nicegui import ui

with ui.row().classes('w-full outline'):
    toggle1 = ui.toggle([1, 2, 3], value=1).props('spread')
with ui.row():
    toggle2 = ui.toggle({1: 'A', 2: 'B', 3: 'C'}).bind_value(toggle1, 'value')

ui.run()
```

here is the check,it seems there is no q-btn-group--spread in the Styles
![0](https://github.com/zauberzeug/nicegui/assets/25096045/2d3830ab-c206-4d96-945e-f5ac7d5535e9)

**Comment by falkoschindler** [comment]
I'm not sure if this is a bug. Does anyone else have an idea what is going on compared to the [Quasar example](https://quasar.dev/vue-components/button-toggle#example--spread-horizontally)?

Luckily, there is a simple workaround by adding the class "w-full":
```py
with ui.row().classes('w-full outline'):
    ui.toggle([1, 2, 3]).props('spread').classes('w-full')
```

**Reply by python-and-novella** [reply]
In quasar's layout `col` whose style is `display:block` ,`spread` works well.
In quasar's layout `row` whose style is `display:flex` ,it works as same as in nicegui's `row` or `column`.
In nicegui,when parent element's style is `display:block` ,it works well.
```python3
from nicegui import ui

with ui.row().classes('w-full outline').style(replace='display:block'):
    toggle1 = ui.toggle([1, 2, 3], value=1).props('spread')
with ui.column().classes('w-full outline').style(replace='display:block'):
    toggle1 = ui.toggle([1, 2, 3], value=1).props('spread')
with ui.element('div').classes('w-full outline'):
    toggle3 = ui.toggle([1, 2, 3], value=1).props('spread')

ui.run()
```
![image](https://github.com/zauberzeug/nicegui/assets/16256910/564135c3-1d64-41b2-93c3-1ade709a654d)

**Reply by falkoschindler** [reply]
Strange, I thought I tested something along those lines... Anyway, thanks for clearing things up! 😀

Here is a similar example using a plain `div` without any "display" setting:
```py
with ui.element().classes('w-full outline'):
    ui.toggle([1, 2, 3]).props('spread')
```

**Reply by python-and-novella** [reply]
@falkoschindler 
Following code works as same as quasar's `row` and `col`, I think it is not a bug of nicegui. Maybe quasar's maintainer could explain it.
```python3
with ui.element().classes('row w-full'):
    ui.toggle([1, 2, 3]).props('spread')
with ui.element().classes('col w-full'):
    ui.toggle([1, 2, 3]).props('spread')
```
